src/main/java/se/liu/ida/hefquin/engine/queryplan/executable/impl/ops/ExecOpRequestTPFatTPFServerWithTranslation.java

```diff
@@ -0,0 +1,152 @@
+package se.liu.ida.hefquin.engine.queryplan.executable.impl.ops;
+
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Set;
+
+import org.apache.jena.graph.NodeFactory;
+import org.apache.jena.sparql.core.Var;
+import org.apache.jena.sparql.engine.binding.Binding;
+import org.apache.jena.sparql.engine.binding.BindingBuilder;
+
+import se.liu.ida.hefquin.engine.data.SolutionMapping;
+import se.liu.ida.hefquin.engine.data.Triple;
+import se.liu.ida.hefquin.engine.data.impl.SolutionMappingImpl;
+import se.liu.ida.hefquin.engine.data.impl.TripleImpl;
+import se.liu.ida.hefquin.engine.federation.TPFServer;
+import se.liu.ida.hefquin.engine.federation.access.FederationAccessException;
+import se.liu.ida.hefquin.engine.federation.access.FederationAccessManager;
+import se.liu.ida.hefquin.engine.federation.access.TPFRequest;
+import se.liu.ida.hefquin.engine.federation.access.TPFResponse;
+import se.liu.ida.hefquin.engine.federation.access.TriplePatternRequest;
+import se.liu.ida.hefquin.engine.federation.access.impl.req.TPFRequestImpl;
+import se.liu.ida.hefquin.engine.federation.access.impl.response.TPFResponseImpl;
+import se.liu.ida.hefquin.engine.federation.access.utils.FederationAccessUtils;
+import se.liu.ida.hefquin.engine.query.BGP;
+import se.liu.ida.hefquin.engine.query.SPARQLGraphPattern;
+import se.liu.ida.hefquin.engine.query.SPARQLGroupPattern;
+import se.liu.ida.hefquin.engine.query.SPARQLUnionPattern;
+import se.liu.ida.hefquin.engine.query.TriplePattern;
+
+/**
+ * Implementation of an operator to request a (complete) TPF from a TPF server.
+ * This implementation handles pagination of the TPF; that is, it requests all
+ * the pages, one after another.
+ */
+public class ExecOpRequestTPFatTPFServerWithTranslation extends ExecOpGenericTriplePatternRequestWithTPF<TPFServer>
+{
+	public ExecOpRequestTPFatTPFServerWithTranslation( final TriplePatternRequest req, final TPFServer fm ) {
+		super( req, fm );
+	}
+
+	@Override
+	protected TPFResponse performPageRequest( final TPFRequest req,
+	                                          final FederationAccessManager fedAccessMgr )
+			throws FederationAccessException
+	{
+		SPARQLGraphPattern reqTranslation = fm.getVocabularyMapping().translateTriplePattern(req.getQueryPattern());
+		if(reqTranslation instanceof TriplePattern) {
+			final TPFResponse res = FederationAccessUtils.performRequest(fedAccessMgr, new TPFRequestImpl((TriplePattern) reqTranslation), fm);
+			final List<Triple> resTranslation = new ArrayList<>();
+			for(final Triple i : res.getPayload()) {
+				BindingBuilder bb = BindingBuilder.create();
+				final Var s = Var.alloc("s");
+				final Var p = Var.alloc("p");
+				final Var o = Var.alloc("o");
+				bb.add(s, i.asJenaTriple().getSubject());
+				bb.add(p, i.asJenaTriple().getPredicate());
+				bb.add(o, i.asJenaTriple().getObject());
+				SolutionMapping sm = new SolutionMappingImpl(bb.build());
```

Comment:
In this part, I think you need to use the variables that are present in the triple pattern from `req.getQueryPattern()`, instead of introducing your own variables. For instance, if the triple pattern is `(?m, ex:name, "Louise")` you should bind  `?m` to the subjects of the triples that you receive.